src/game/Unit.cpp

```diff
@@ -2499,54 +2521,34 @@ MeleeHitOutcome Unit::RollMeleeOutcomeAgainst(const Unit* pVictim, WeaponAttackT
                 }
                 else
                 {
-                    DEBUG_FILTER_LOG(LOG_FILTER_COMBAT, "RollMeleeOutcomeAgainst: BLOCK <%d, %d)", sum - tmp, tmp);
+                    DEBUG_FILTER_LOG(LOG_FILTER_COMBAT, "RollMeleeOutcomeAgainst: BLOCK <%d, %d)", sum - block_chance, block_chance);
                     return MELEE_HIT_BLOCK;
                 }
             }
         }
     }
 
-    // Max 40% chance to score a glancing blow against mobs that are higher level
-    // (can do only players and pets and not with ranged weapon)
-    if ((GetTypeId() == TYPEID_PLAYER || ((Creature*)this)->IsPet())
-        && pVictim->GetTypeId() != TYPEID_PLAYER && !((Creature*)pVictim)->IsPet()
-        && getLevel() < pVictim->getLevel() && attType != RANGED_ATTACK && !SpellCasted)
-    {
-        tmp = 1000 + ((attackerWeaponSkill < attackerMaxSkillValueForLevel)
-            ? (victimDefenseSkill - attackerWeaponSkill) * 200
-            : (victimDefenseSkill - attackerMaxSkillValueForLevel) * 200);
-
-        if (roll < (tmp = (tmp > 4000) ? 4000 : tmp))
-        {
-            DEBUG_FILTER_LOG(LOG_FILTER_COMBAT, "RollMeleeOutcomeAgainst: GLANCING <%d, %d)", sum - tmp, tmp);
-            return MELEE_HIT_GLANCING;
-        }
-    }
-
-    if (crit_chance > 0)
+    if (crit_chance > 0 && roll < (sum += crit_chance))
```

Comment:
I'd ask why that was being handled in unit but that is down to me for not having checked there before applying the other patch. This may have affected crit chance, indeed. but not in the volumes you speak of.